docs/language/contracts.md

```diff
@@ -261,12 +284,107 @@ let code = "70756220636f6e...".decodeHex()
 // `code` has type `[UInt8]`
 
 let signer: Account = ...
-signer.setCode(
-    code,
+signer.contracts.add(
+    name: "Test"
+    code: code,
     message: "I'm a new contract in an existing account"
 )
 ```
 
+#### Updating a Deployed Contract
+
+> 🚧 Status: Updating contracts is **experimental**.
+
+A deployed contract can be updated using the `update__experimental` function:
+
+  ```cadence
+  fun update__experimental(name: String, code: [UInt8]): DeployedContract
+  ```
+
+  Updates the code for the contract/contract interface in the account.
+
+  The `code` parameter is the UTF-8 encoded representation of the source code.
+  The code must contain exactly one contract or contract interface,
+  which must have the same name as the `name` parameter.
+
+  Does **not** run the initializer of the contract/contract interface again.
+  The contract instance in the world state stays as is.
+
+  Fails if no contract/contract interface with the given name exists in the account,
+  if the given code does not declare exactly one contract or contract interface,
+  or if the given name does not match the name of the contract/contract interface declaration in the code.
+
+  Returns the deployed contract for the updated contract.
+
+For example, assuming that a contract named `Test` is already deployed to the account
+and it should be updated with the following contract code:
+
+```cadence,file=test-contract.cdc
+pub contract Test {
+    pub let message: String
+
+    init(message: String) {
+        self.message = message
+    }
+}
+```
+
+The contract can be updated as follows:
+
+```cadence,file=account-contracts-update.cdc
+// Decode the hex-encoded source code into a byte array
+// using the built-in function `decodeHex`.
+//
+// (The ellipsis ... indicates the remainder of the string)
+//
+let code = "70756220636f6e...".decodeHex()
+
+// `code` has type `[UInt8]`
+
+let signer: Account = ...
+signer.contracts.update__experimental(name: "Test", code: code)
+```
```

Comment:
I think this section probably could use a bigger example showing both versions of the `Test` contract (before and after), and which kind of upgrade is safe and which kind isn't. If you don't have time to add it and want to merge this now though, I can write that example.

Reply:
Yeah, I agree that a more extensive example would be useful, I mainly only documented the API. 
I don't really have the time right now to write that, really just trying to get the PR in, it's been open for over a month now. If you could maybe add that later, once you have time, that would be great! 🙏

Reply:
Sounds good to me! Yeah it is good to just get the PR in and update later. 👍